jobs/integration/test_opa.py

```diff
@@ -0,0 +1,439 @@
+                storage_pool_name,
+                "kubernetes",
+                "-m",
+                tools.k8s_connection,
+                "storage-class=mystorage",
+            )
+            yield storage_pool_name
+        finally:
+            log("Removing storage pool")
+            await tools.run(
+                "juju",
+                "remove-storage-pool",
+                "-m",
+                tools.k8s_connection,
+                storage_pool_name,
+            )
+
+    @pytest.fixture(scope="class")
+    async def update_status_interval(self, tools, k8s_model):
+        interval = await tools.run(
+            "juju",
+            "model-config",
+            "-m",
+            tools.k8s_connection,
+            "update-status-hook-interval",
+        )
+        interval = interval[0].strip()
+        try:
+            log("Decrease update status hook interval")
+            await tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                tools.k8s_connection,
+                "update-status-hook-interval=5s",
+            )
+            yield
+        finally:
+            await tools.run(
+                "juju",
+                "model-config",
+                "-m",
+                tools.k8s_connection,
+                f"update-status-hook-interval={interval}",
+            )
+
+    @pytest.fixture(scope="class")
+    async def opa_controller_manager(self, model, k8s_model, tools):
+        try:
+            log("Deploying the webhook charm")
+            webhook = await k8s_model.deploy(
+                "gatekeeper-controller-manager",
+                channel=tools.charm_channel,
+                trust=True,
+            )
+            yield webhook
+        finally:
+            log("Deleting the gatekeeper charm")
+            await tools.run(
+                "juju",
+                "remove-application",
+                "-m",
+                tools.k8s_connection,
+                "--force",
+                "gatekeeper-controller-manager",
+            )
+            await self.wait_for_app_removed("gatekeeper-controller-manager", k8s_model)
+
+    @pytest.fixture(scope="class")
+    async def opa_audit(self, model, k8s_model, tools, storage_pool):
+        try:
+            log("Deploying the audit charm")
+            audit = await k8s_model.deploy(
+                "gatekeeper-audit",
+                channel=tools.charm_channel,
+                trust=True,
+                storage={"audit-volume": {"pool": storage_pool}},
+                config={"audit-interval": 1},
+            )
+            yield audit
+        finally:
+            log("Deleting the audit charm")
+            await tools.run(
+                "juju",
+                "remove-application",
+                "-m",
+                tools.k8s_connection,
+                "--force",
+                "--destroy-storage",
+                "gatekeeper-audit",
+            )
```

Comment:
```suggestion
            await k8s_model.remove_application(audit.name)
```

Reply:
This does not remove storage, I thought it's simpler to run the juju command rather than having to fetch the used storages and free them before to removing the storage-pool

Reply:
ok, how about 

```python
k8s_model.remove_application(audit.name, destroy_storage=True, force=True)
```

Reply:
https://github.com/juju/python-libjuju/blob/master/juju/model.py#L1008